### PR TITLE
Add procedure to add passphrase to SSH key used for REX

### PR DESCRIPTION
**Pull request by adamlazik1**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2170031

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.9/Katello 4.11 (planned Satellite 6.15)
* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* [x] Foreman 3.4/Katello 4.6 (EL8 only)
* [x] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* We do not accept PRs for Foreman older than 3.3.

### CODE REVIEW

File: guides/common/modules/proc_creating-new-ssh-keys-for-remote-execution.adoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[id="creating-new-ssh-keys-for-remote-execution_{context}"]
+= Creating new SSH keys for remote execution
+
+By default, {Project} uses a non-passphrase protected SSH key to execute remote jobs on hosts.
+You can create additional passphrase protected or non-passphrase protected SSH keys to authenticate for remote execution jobs on hosts.
```
Comment (adamruzicka) [inline guides/common/modules/proc_creating-new-ssh-keys-for-remote-execution.adoc:5]: The word `additional` could be misleading in this context. It makes it read as if you could use multiple keys at once, while a smart proxy expects exactly one key.

File: guides/common/modules/proc_creating-new-ssh-keys-for-remote-execution.adoc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+.Procedure
+. On your {ProjectServer} or {SmartProxyServer}, create a new SSH key:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# ssh-keygen
```
Comment (adamruzicka) [inline guides/common/modules/proc_creating-new-ssh-keys-for-remote-execution.adoc:12]: Just generating the key is not enough. The key in questio has to be placed into an expected location (and thus replace the old key) so something like `ssh-keygen -f ~foreman-proxy/.ssh/id_rsa_foreman_proxy` would be appropriate

Alternatively, you can change the expected location by passing a combaination of `--foreman-proxy-plugin-remote-execution-script-ssh-identity-dir=$DIR` and `--foreman-proxy-plugin-remote-execution-script-ssh-identity-file=$KEY` to the installer

**Review by adamruzicka — CHANGES_REQUESTED** [review]

**Comment by adamlazik1** [comment]
Since there can be only one key I opted for adding a passphrase to existing key instead. This way there is no additional action required for existing hosts.

File: guides/common/modules/proc_adding-a-passphrase-to-ssh-key-used-for-remote-execution.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+.Procedure
+* On your {ProjectServer} or {SmartProxyServer}, use `ssh-keygen` to add a passphrase to your SSH key:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# ssh-keygen -p -f ~foreman-proxy/.ssh/id_rsa_foreman_proxy
+----
+
+.Next steps
+* Users now must authenticate with a passphrase when running remote execution jobs on hosts.
```
Comment (adamruzicka) [inline guides/common/modules/proc_adding-a-passphrase-to-ssh-key-used-for-remote-execution.adoc:16]: Nit: You use the passphrase to unlock the key and then authenticate with the unlocked key. You don't really authenticate with the passphrase

File: guides/common/modules/proc_adding-a-passphrase-to-ssh-key-used-for-remote-execution.adoc [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+By default, {SmartProxy} uses a non-passphrase protected SSH key to execute remote jobs on hosts.
+You can protect the SSH key with a passphrase by following this procedure.
+
+.Procedure
+* On your {ProjectServer} or {SmartProxyServer}, use `ssh-keygen` to add a passphrase to your SSH key:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# ssh-keygen -p -f ~foreman-proxy/.ssh/id_rsa_foreman_proxy
```
Comment (adamruzicka) [inline guides/common/modules/proc_adding-a-passphrase-to-ssh-key-used-for-remote-execution.adoc:12]: Should this mention that this is just the default path?

Comment (adamlazik1) [reply guides/common/modules/proc_adding-a-passphrase-to-ssh-key-used-for-remote-execution.adoc:12]: I was considering something like "if you changed the location of your ssh key, then use the new location" but just sounds too obvious, so I would probably opt for not mentioning that unless there are objections.

File: guides/common/modules/proc_adding-a-passphrase-to-ssh-key-used-for-remote-execution.adoc (hunk 3 above)
Comment (adamlazik1) [reply guides/common/modules/proc_adding-a-passphrase-to-ssh-key-used-for-remote-execution.adoc]: Changed wording.

**Review by adamruzicka — APPROVED** [review]
LGTM

**Comment by adamlazik1** [comment]
Merged and cherry-picked:
   3cb5aecad3..a2c0d0b37d  3.10 -> 3.10
   3569e9d026..5f13a91957  3.3 -> 3.3
   d338e4af51..ce59624bd8  3.4 -> 3.4
   617e5f4327..9786906eae  3.5 -> 3.5
   1dd717a432..731bb132fe  3.6 -> 3.6
   f84b27533b..144f1eab68  3.7 -> 3.7
   a86277a422..33c2d9d1ce  3.8 -> 3.8
   dec4d737df..902ba48156  3.9 -> 3.9

**Pull request merged**